src/emulate.c

```diff
@@ -348,9 +350,11 @@ FORCE_INLINE bool insn_is_misaligned(uint32_t pc)
     );
 }
 
-/* can-branch information for each RISC-V instruction */
+/* can-branch and insn_len information for each RISC-V instruction */
 enum {
-#define _(inst, can_branch, reg_mask) __rv_insn_##inst##_canbranch = can_branch,
+#define _(inst, can_branch, insn_len, reg_mask) \
+    __rv_insn_##inst##_len = insn_len,          \
+    __rv_insn_##inst##_canbranch = can_branch,
     RV_INSN_LIST
```

Comment:
This modification of single `enum` would provide `RV_INSN_LIST` an unexpected value. Instead, you must use two 'enum'. The term `RV_INSN_LIST` only appears once.

Reply:
However, `enum` is outside the define,
```c
enum {
#define _(inst, can_branch, insn_len, reg_mask) \
    __rv_insn_##inst##_len = insn_len,
    RV_INSN_LIST
#undef _
};
```
Do you means a `enum` inside the `enum`?

Reply:
Ouch. I misinterpreted the code. No visible concern here.